### PR TITLE
Core: refactor some loading mechanisms

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
on its own, not much

## How was this tested?
together with a new component that comes seperate

### CODE REVIEW

**Review by black-sliver — CHANGES_REQUESTED** [review]
This breaks the following use-case:
`python Launcher.py Generate`

**Comment by black-sliver** [comment]
I think the new code handles all use-cases we now have, with one exception:
Launching it with an invalid argument would not open to GUI previously. I don't have a strong opinion, but printing a message would be nice (in addition to opening the GUI), I think

With the current refactor, I think identify can be changed to only return 2 values - the original code would allow launching everything from the AP folder iirc, but that's not required anymore.

**Comment by Berserker66** [comment]
It is reduced to returning 2 values already, but the change in the branch is not showing up in this PR.

**Comment by Berserker66** [comment]
well, merging main into the branch again seems to have refreshed github

**Comment by black-sliver** [comment]
Do we want to log/print/show a message if the argument was ignored/not found? Once we decided that, I think this can be merged

File: Launcher.py [hunk 1]
```diff
@@ -221,23 +224,34 @@ def component_action(button):
...
         args = {}
 
     if "Patch|Game|Component" in args:
-        file, component, _ = identify(args["Patch|Game|Component"])
+        file, component = identify(args["Patch|Game|Component"])
         if file:
             args['file'] = file
         if component:
             args['component'] = component
+        if not file or component:
```
Comment (black-sliver) [inline Launcher.py:248]: pretty sure it should be this?
```suggestion
        if not component:
```

Comment (Berserker66) [reply Launcher.py:248]: You know, I didn't test the logging addition. Of course the thing I didn't test I'd slip up.

**Review by black-sliver — APPROVED** [review]

**Pull request merged**